TCAT/Model/Route.swift

```diff
@@ -224,38 +142,20 @@ class Route: NSObject, JSONDecodable {
     
     /** Calculate travel distance from location passed in to first route summary object and updates travel distance of route
      */
-    func calculateTravelDistance(fromLocation location: CLLocationCoordinate2D) {
-        guard let firstRouteSummary = routeSummary.first else {
+    func calculateTravelDistance(fromDirection directions: [Direction]) {
+        // first route option stop is the first bus stop in the route
+        guard let firstRouteOptionsStop = directions.first?.type == .walk ? directions[1] : directions.first else {
             return
         }
         
-        let fromLocation = CLLocation(latitude: location.latitude, longitude: location.longitude)
-        let endLocation = CLLocation(latitude: firstRouteSummary.location.latitude, longitude: firstRouteSummary.location.longitude)
+        let fromLocation = CLLocation(latitude: startCoords.latitude, longitude: startCoords.longitude)
+        let endLocation = CLLocation(latitude: firstRouteOptionsStop.startLocation.latitude, longitude: firstRouteOptionsStop.startLocation.longitude)
         
-        var distance = fromLocation.distance(from: endLocation)
-        distance = distance * 0.000621371192 // convert meters to miles
-        self.travelDistanceDelegate?.travelDistanceUpdated(withDistance: distance)
-    }
-    
-    /** Update pin type of the last routeSummaryObject if routeSummaryObject has the same name as the user searched for
-     * OR update name and pin type if last routeSummaryObject is "End"
-     */
-    func updateEndingDestination(_ place: Place) {
-        if let lastRouteSummaryObject = routeSummary.last {
-            if lastRouteSummaryObject.name == place.name || lastRouteSummaryObject.name == "End" {
-                let type = place is BusStop ? PinType.stop : PinType.place
-                lastRouteSummaryObject.type = type
-                
-                // Remove “End” if 2nd-to-last stop has same name as ending destination
-                let secondToLastStop = routeSummary.count-2
-                if lastRouteSummaryObject.name == "End" && routeSummary.count > 2 && routeSummary[secondToLastStop].name.lowercased() == place.name.lowercased() {
-                    routeSummary.removeLast()
-                    routeSummary.last?.removeNextDirection() // remove 2nd-to-last stop next direction (so it behaves like ending stop)
-                } else if(lastRouteSummaryObject.name == "End") {
-                    lastRouteSummaryObject.updateName(from: place)
-                }
-            }
-        }
+        let numberOfMetersInMile = 1609.34
+        let distanceInMeters = fromLocation.distance(from: endLocation)
```

Comment:
I'd like to centralize conversions if possible, hmm

Reply:
yeah maybe we can think of a class later to put that in or make it an extension of double?